median/aggregated_attribute_observation.go

```diff
@@ -0,0 +1,48 @@
+package median
+
+import (
+	"cmp"
+	"math/big"
+	"slices"
+	"sort"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+)
+
+type aggregatedAttributedObservation struct {
+	Timestamp       uint32
+	Observers       [32]commontypes.OracleID
+	Observations    []*big.Int
+	JuelsPerFeeCoin *big.Int
+}
+
+func aggregate(observations []median.ParsedAttributedObservation) *aggregatedAttributedObservation {
+	// defensive copy
+	n := len(observations)
+	observations = append([]median.ParsedAttributedObservation{}, observations...)
+
+	aggregated := &aggregatedAttributedObservation{Observations: make([]*big.Int, len(observations))}
+
+	slices.SortFunc(observations, func(a, b median.ParsedAttributedObservation) int {
+		return cmp.Compare(a.Timestamp, b.Timestamp)
+	})
+	aggregated.Timestamp = observations[n/2].Timestamp
+
+	// get median juelsPerFeeCoin
+	sort.Slice(observations, func(i, j int) bool {
```

Comment:
`slice.SortFunc` should be more efficient in these other two cases, since `sort.Slice` has reflection.

Reply:
Thanks, so used to not having new enough go to use that.